### PR TITLE
[MapRouletteUploadCommand] Fix custom challenge name bug

### PR DESCRIPTION
**Pull request by danielduhh**
### Description:

This fixes an issue reported by @dcastrowa where the `MapRouletteUploadCommand` was no longer able to set custom Challenge names. 

#222 brought in the ability to prepend country display names in each challenge, however, it set each challenge name to the `challengeName` property of the `Task` (`Task.getChallengeName()`). Since changing the `Challenge` name (via `Challenge.setName()`) does not propagate to its Tasks, the `Challenge` name sent to the API did not include the custom name in the configuration.

### Potential Impact:

This will allow us to use custom challenge names when uploading to MapRoulette

### Unit Test Approach:

I added a couple unit tests that run the command using a custom challenge configuration. I also tested the command uploads to MR.

### Test Results:

Unit tests pass.

Custom challenge names now show up in MR

Screen Shot 2020-02-17 at 3 36 11 PM

### CODE REVIEW

File: src/test/resources/org/openstreetmap/atlas/checks/maproulette/checksConfig.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    "challenge": {
+      "name": "Spiky Buildings"
+    }
+  },
+  "SomeOtherCheck": {
+    "challenge": {
+      "name": "Intersecting Lines"
+    }
+  }
+}
```
Comment (seancoulter) [inline src/test/resources/org/openstreetmap/atlas/checks/maproulette/checksConfig.json]: Nit: should be newline here

File: src/test/java/org/openstreetmap/atlas/checks/maproulette/MapRouletteUploadCommandTest.java [hunk 2]
```diff
@@ -83,6 +116,30 @@ public void writeFiles()
...
 
+    /**
+     * Similar to runAndTest, however this function will return a {@link TestMapRouletteConnection}
+     *
+     * @param additionalArguments
+     *            String[] of extra arguments for {@link MapRouletteUploadCommand}, to add to the
+     *            data i/o locations and server config
+     * @return TestMapRouletteConnection
+     */
+    private TestMapRouletteConnection run(final String[] additionalArguments)
```
Comment (seancoulter) [inline src/test/java/org/openstreetmap/atlas/checks/maproulette/MapRouletteUploadCommandTest.java]: runAndTest could call this method to cut down on some duplication

**Comment by seancoulter** [comment]
Thanks for fixing this! Had a couple small requests.

File: src/test/java/org/openstreetmap/atlas/checks/maproulette/MapRouletteUploadCommandTest.java (hunk 2 above)
Comment (danielduhh) [reply src/test/java/org/openstreetmap/atlas/checks/maproulette/MapRouletteUploadCommandTest.java]: Yup, good call. This was handled in the latest commit

File: src/test/java/org/openstreetmap/atlas/checks/maproulette/MapRouletteUploadCommandTest.java [hunk 3]
```diff
@@ -57,7 +60,39 @@ public void testCountryFilter()
...
+        final Set<Project> projects = connection.uploadedProjects();
+
+        final List<String> challengeNames = projects.stream().flatMap(project -> connection
+                .challengesForProject(project).stream().map(Challenge::getName))
+                .collect(Collectors.toList());
+        Collections.sort(challengeNames);
+
+        Assert.assertEquals("Canada - Spiky Buildings", challengeNames.get(0));
+        Assert.assertEquals("Mexico, Belize - Intersecting Lines", challengeNames.get(1));
+        Assert.assertEquals("United States - Address Point Match", challengeNames.get(2));
```
Comment (dcastrowa) [inline src/test/java/org/openstreetmap/atlas/checks/maproulette/MapRouletteUploadCommandTest.java]: nice testing! cool to see it java and how I can try and create testing in python.

**Pull request merged**